bot/admin/test/core/src/main/kotlin/TestPlanService.kt

```diff
@@ -144,8 +146,12 @@ object TestPlanService {
             }
         }
 
-        // save the test plan execution into the database
-        testPlanDAO.saveTestExecution(exec.copy(status = TestPlanExecutionStatus.COMPLETE))
+        if (nbErrors != 0) {
+            exec.nbErrors = nbErrors
```

Comment:
It seems odd to modify `nbErrors` to `var` only for this case. Why don't you return the copy below which has the right error count and keep `nbErrors` as `val` ?

Reply:
Indeed, now only "copy()" is used to update value in MongnDB. On your advice, I've also store the exec copy into another variable to return the right updated one.